---
layout: default
title: Home
nav_order: 0
permalink: /
---

<html>
<img
src="https://thebrownlab.github.io/phylofisher-pages/assets/images/TREE-PF-LOGO.png"
alt="fisher.py Scheme"
height="75%"
width="100%" 
class="center"/>

<!-- Badges -->
<div align="center">
    <!-- Citation Badge -->
    <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3001365"><img src="https://img.shields.io/badge/dynamic/json?color=blue&label=Citations&query=times_cited&url=https%3A%2F%2Fmetrics-api.dimensions.ai%2Fdoi%2F10.1371%2Fjournal.pbio.3001365&style=flat-square" /></a>
    <!-- Link to GitHub Badge -->
    <a href="https://github.com/TheBrownLab/PhyloFisher"><img src="https://img.shields.io/badge/-GitHub-455149?logo=github&logoColor=white&link&style=flat-square" /></a>
    <!-- Install with Conda Badge -->
    <a href="https://anaconda.org/bioconda/phylofisher"><img src="https://anaconda.org/bioconda/phylofisher/badges/installer/conda.svg" /></a>
    <!-- Conda Downloads Badge -->
    <a href="https://anaconda.org/bioconda/phylofisher"><img src="https://anaconda.org/bioconda/phylofisher/badges/downloads.svg" /></a>
    <!-- Conda Latest Release Badge -->
    <a href="https://anaconda.org/bioconda/phylofisher"><img src="https://anaconda.org/bioconda/phylofisher/badges/latest_release_date.svg" /></a>
    <!-- License Badge -->
    <a href="https://opensource.org/licenses/MIT"><img src="https://anaconda.org/bioconda/phylofisher/badges/license.svg" /></a>
</div>

<hr>

<!-- Introduction -->
<p>
    PhyloFisher is a software package, written in Python3, that contains a protocol designed for phylogenomic dataset assembly and data exploration. This software package aids in the construction and curation of protein sequence-based phylogenomic datasets, conducts post-assembly analyses, and allows visualization of the results. In addition, PhyloFisher currently includes a manually curated starting dataset of 240 proteins from 304 eukaryotic taxa representing the full breadth of known diversity in the eukaryotic tree of life. Importantly, this dataset also includes identified paralogs of each of the 240 proteins from all investigated taxa which is crucial for the identification of probable orthologs. Although PhyloFisher includes this pan-eukaryotic dataset, the tool is flexible and can work with any dataset consisting of protein sequences derived from eukaryotes. The combination of all of the foregoing features makes PhyloFisher a broadly-useful, user-friendly software tool for sophisticated phylogenomic analyses of eukaryotes.
</p>

<h2>How to Cite</h2>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Tice AK*, Žihala D*, Pánek T*, Jones RE", Salomaki E", Nenarokov S, Burki F, Eliáš M, Eme L, Roger AJ, Rokas A, Shen X, Strassert JFH, Kolísko M, Brown MW. 2021. PhyloFisher: A phylogenomic package for resolving eukaryotic relationships. [PLoS Biology. 19(8): e3001365](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3001365) 
    </p>
    <p>* & " = Contributed equally</p>

<h2>Authors</h2>

    <h3>Software Developers</h3>
        <ul>
            <li>David Žihala, PhD - <a href="https://scholar.google.com/citations?user=jTL1c6QAAAAJ&hl=en">Google Scholar</a></li>
            <li>Robert Jones - <a href="https://scholar.google.com/citations?user=pQeUuoIAAAAJ&hl=en">Google Scholar</a></li>
            <li>Serafim Nenarokov - <a href="https://scholar.google.ru/citations?user=maAnfhQAAAAJ&hl=en">Google Scholar</a></li>
        </ul>
        
    <h3>Tree Thinkers</h3>
        <ul>
            <li>Alexander K Tice, PhD - <a href="https://scholar.google.cz/citations?user=4jR5i0sAAAAJ&hl=en">Google Scholar</a></li>
            <li>Eric Salomaki, PhD - <a href="https://scholar.google.com/citations?user=u0M3ySEAAAAJ&hl=en">Google Scholar</a></li>
        </ul>

    <h3>Core PI Team</h3>
        <ul>
            <li>Matthew W. Brown, PhD - <a href="https://scholar.google.com/citations?user=UvN4AQsdfygC&hl=en">Google Scholar</a></li>
            <li>Martin Kolisko, PhD - <a href="https://scholar.google.com/citations?user=hfRmRWcAAAAJ&hl=en">Google Scholar</a></li>
            <li>Fabien Burki, PhD - <a href="https://scholar.google.com/citations?user=Uhvos9EAAAAJ&hl=en">Google Scholar</a></li>
            <li>Marek Eliáš, PhD - <a href="https://www.osu.eu/marek-elias/20385/">Lab Website</a></li>
            <li>Laura Eme, PhD - <a href="https://scholar.google.com/citations?user=8FOe6yMAAAAJ&hl=en">Google Scholar</a></li>
            <li>Tomáš Pánek, PhD - <a href="https://scholar.google.com/citations?hl=en&user=XtWBV9MAAAAJ&fbclid=IwAR1ybDIhzeyeCkLcD1vhinvjlbyYDZWpbWW1Al1Nw2vq0VsU_1eGDAsa4rU">Google Scholar</a></li>
            <li>Andrew Roger, PhD - <a href="https://scholar.google.com/citations?user=Dm-pAawAAAAJ&hl=en">Google Scholar</a></li>
            <li>Antonis Rokas, PhD - <a href="https://scholar.google.com/citations?user=OvAV_eoAAAAJ&hl=en">Google Scholar</a></li>
            <li>Xing-Xing Shen, PhD - <a href="https://scholar.google.com/citations?user=aNqPBsEAAAAJ&hl=en">Google Scholar</a></li>
        </ul>
