---
layout: default
title: References
nav_order: 101
permalink: /references
---

# References

<html>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Ali, R. H., Bogusz, M., & Whelan, S. (2019). Identifying clusters of high confidence homologies in multiple sequence alignments. Molecular Biology and Evolution, 36(10), 2340–2351. https://doi.org/10.1093/ molbev/msz142
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Altschul, S. F., Gish, W., Miller, W., Myers, E. W., & Lipman, D. J. (1990). Basic local alignment search tool. J. Mol. Biol., 215(3), 403–410. https://doi.org/10.1016/S0022-2836(05)80360-2 Brown, M. W., Heiss, A. A., Kamikawa, R., Inagaki, Y., Yabuki, A., Tice, A. K., Shiratori, T., Ishida, K.-I., Hashimoto, T., Simpson, A. G. B., & Roger, A. J. (2018). Phylogenomics places orphan protistan lineages in a novel eukaryotic super-group. Genome Biology and Evolution, 10(2), 427–433. https: //doi.org/10.1093/gbe/evy014
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Buchfink, B., Xie, C., & Huson, D. H. (2015). Fast and sensitive protein alignment using DIAMOND. Nature Methods, 12(1), 59–60. https://doi.org/10.1038/nmeth.3176
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Capella-Gutiérrez, S., Silla-Martínez, J. M., & Gabaldón, T. (2009). trimAl: A tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics, 25(15), 1972–1973. https://doi.org/10.1093/bioinformatics/btp348
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Chen, F., Mackey, A. J., Stoeckert, C. J., Jr, & Roos, D. S. (2006). OrthoMCL-DB: Querying a comprehensive multi-species collection of ortholog groups. Nucleic Acids Research, 34, D363–D368. https://doi.org/10.1093/nar/gkj123
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Criscuolo, A., & Gribaldo, S. (2010). BMGE (block mapping and gathering with entropy): A new software for selection of phylogenetic informative regions from multiple sequence alignments. BMC Evolutionary Biology, 10(1), 210. https://doi.org/10.1186/1471-2148-10-210
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Fu, L., Niu, B., Zhu, Z., Wu, S., & Li, W. (2012). CD-HIT: Accelerated for clustering the next-generation sequencing data. Bioinformatics, 28(23), 3150–3152. https://doi.org/10.1093/bioinformatics/bts565
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Huerta-Cepas, J., Serra, F., & Bork, P. (2016). ETE 3: Reconstruction, Analysis, and Visualization of Phylogenomic Data. Molecular Biology and Evolution, 33(6), 1635–1638. https://doi.org/10.1093/molbev/msw046
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Inagaki, Y., Susko, E., Fast, N. M., & Roger, A. J. (2004). Covarion Shifts Cause a Long-Branch Attraction Artifact That Unites Microsporidia and Archaebacteria in EF-1alpha Phylogenies. Molecular Biology and Evolution, 21(7), 1340–1349. https://doi.org/10.1093/molbev/msh130
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Katoh, K., & Standley, D. M. (2013). MAFFT multiple sequence alignment software version 7: Improvements in performance and usability. Molecular Biology and Evolution, 30(4), 772–780. https://doi.org/10.1093/molbev/mst010
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Lopez, P., Casane, D., & Philippe, H. (2002). Heterotachy, an Important Process of Protein Evolution. Molecular Biology and Evolution, 19(1), 1–7. https://doi.org/10.1093/ oxfordjournals.molbev.a003973
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Mistry, J., Finn, R. D., Eddy, S. R., Bateman, A., & Punta, M. (2013). Challenges in homology search: HMMER3 and convergent evolution of coiled-coil regions. Nucleic Acids Research, 41(12), e121–e121.https://doi.org/10.1093/nar/gkt263
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Price, M. N., Dehal, P. S., & Arkin, A. P. (2010). FastTree 2 - approximately maximum-likelihood trees for large alignments. PLoS ONE, 5(3), 1–10. https://search.ebscohost.com/login.aspx?direct=true&db=a9h&AN=56440094&site=eds-live&scope=site&custid=magn1307
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Salomaki, E.D., Terpis, K.X., Rueckert,  Michael Kotyk, Zuzana Kotyková Varadínová, Ivan Čepička, Christopher E. Lane & Martin Kolisko. (2021). Gregarine single-cell transcriptomics reveals differential mitochondrial remodeling and adaptation in apicomplexans. BMC Biol. 19(77). https://doi.org/10.1186/s12915-021-01007-2
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30(9), 1312–1313. https://doi.org/10.1093/bioinformatics/btu033
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Susko, E., Field, C., Blouin, C., & Roger, A. J. (2003). Estimation of rates-across-sites distributions in phylogenetic substitution models. Systematic Biology, 52(5), 594–603. https://doi.org/10.1080/10635150390235395
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Susko, E., Lincker, L., & Roger, A. J. (2018). Accelerated estimation of frequency classes in site-heterogeneous profile mixture models. Molecular Biology and Evolution, 35(5), 1266–1283. https://doi.org/10.1093/molbev/msy026
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Susko, E., & Roger, A. J. (2007). On Reduced Amino Acid Alphabets for Phylogenetic Inference. Molecular Biology and Evolution, 24(9), 2139–2150. https://doi.org/10.1093/molbev/msm144
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Tice, A. K., Shadwick, L. L., Fiore-Donno, A. M., Geisen, S., Kang, S., Schuler, G. A., Spiegel, F. W., Wilkinson, K. A., Bonkowski, M., Dumack, K., Lahr, D. J. G., Voelcker, E., Clauß, S., Zhang, J., & Brown, M. W. (2016). Expansion of the molecular and morphological diversity of acanthamoebidae (centramoebida, amoebozoa) and identification of a novel life cycle type within the group. Biology Direct, 11(1), 69. https://doi.org/10.1186/s13062-016-0171-0
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Whelan, S., Irisarri, I., & Burki, F. (2018). PREQUAL: Detecting non-homologous characters in sets of unaligned homologous sequences. Bioinformatics, 34(22), 3929–3930. https://doi.org/10.1093/bioinformatics/bty448
    </p>
    <p align="justify", style="text-indent: -36px; padding-left: 36px;">
        Zhang, C., Rabiee, M., Sayyari, E., & Mirarab, S. (2018). ASTRAL-III: Polynomial time species tree reconstruction from partially resolved gene trees. BMC Bioinformatics, 19(6), 153. https://doi.org/10.1186/s12859-018-2129-y
    </p>
</html>